### ISSUE TITLE
E2EE : Client creates encrypted files, messes with real one

### ISSUE THREAD

**Issue by GLLM** [nextcloud/desktop#288]
Hi all !

NC server 13.0.3 running on Ubuntu 16.04 LTS + MySQL
Client E2E tech preview running on Windows 7 64bits
Android 3.0.2 client running on Android 8.0 on a Samsung Galaxy S8+

When I add files in the local (PC) folder which is to be E2E encrypted on the server, I get really weird behavior when I force-sync my Android phone :
* at first the file is uplaod and E2EE properly
* then it seems to do some weird sync stuff : remove then adds again the files which were on the server prior to the file addition
* then some files are deleted (from Android as well as from local PC)
* then I get the below : it creates really weird files on Android

![image](https://user-images.githubusercontent.com/3989573/39828022-e52f98f6-53b9-11e8-9f17-88e873c46ced.png)

tl;dr : I'm happy to test and preview the E2EE feature, yet hopefully I just add, change & remove only test files otherwise I woud have lost data.

I'm not sure howI can help.
G_LL_M

**Comment by camilasan** [comment]
We will provide a new tech preview soonish. We have a much newer version of the e2ee (#251) merged into master. I believe this problem has been fixed but just in case cc @tcanabrava @dantti.

**Comment by dantti** [comment]
I believe it's been fixed as the desktop client store files with UUIDs as names now, of course this might be created by the Android app which I dunno if it's using UUIDs already, but the behavior that deleted and downloaded the file again should be fixed now.

**Comment by GLLM** [comment]
Great !
Can't wait to test the next tech preview version :)
Will update the issue when done !

Thanks all !

**Comment by tcanabrava** [comment]
Should be fixed.

**Comment by GLLM** [comment]
Has "soonish" been postponed or rescheduled ? ;-)

**Comment by camilasan** [comment]
:)

but soonish became today: https://download.nextcloud.com/desktop/daily/Windows/.

**Comment by GLLM** [comment]
hey !
I just installed the 28.06 daily.
I had to enter the 12 words for the decryption.
It seemed to have failed to decrpt it locally, but I do not knowwhy because when I look at the mnemonic, they seem fine.

Here is the error messages i get :
![image](https://user-images.githubusercontent.com/3989573/42084363-bc41e9de-7b8d-11e8-98fa-ae4d7d98c14e.png)

Also, I cannot remove *any* file fomr my E2EE test folder (eiehter from Android or from Windows). I'm kinda stuck with the actual files + all the crappy files the E2E generated (see the issue title)

Do I need to wait for a NC server update so that the E2E works again ?
Thanks
GLLM

NC : 13.0.4 on ubuntu 16.04 LTS - MySQL
Windows client : Nextcloud-2.5.0.0-Debug-20180628
Android client : 3.2 stable

**Comment by er-vin** [comment]
It has been a while. I think this is fixed now as in "we probably can't reach that state anymore". For the stale files you pointed out I'm not sure we can do anything about them anymore I suspect they ended up in a broken state during the early tech preview.

I will close this one for now, but please if you manage to corrupt files again like you initially reported with upcoming 2.7 releases feel free to open an issue with an updated reproduction procedure.

**Comment by GLLM** [comment]
HI @er-vin 

no pb, I resort to stop using/hoping for the E2E encryption of files in Nextcloud ... especially since this feature (or feature-to be, maybe one day), is not advertised at all by Nextcloud.

Thanks

**Comment by KopfKrieg** [comment]
> especially since this feature (or feature-to be, maybe one day), is not advertised at all by Nextcloud.

Well, erm… 
* https://nextcloud.com/blog/nextcloud-13-brings-secure-file-sync-and-collaboration-to-the-next-level/
* https://nextcloud.com/endtoend/

**Comment by GLLM** [comment]
@KopfKrieg  : well it's 2y ago & NC 13 ... and I have never heard anything about E2E encryption in NC ever since.
When this was released, I installed it, tried it. Had many bugs, file which could not be decipher. Other which could not be visible.
I removed it 1.5y later because it was not mature and full-featured enough. So I've been waiting for something more stable and usable.
I think I can probably find the github issue which explained that quite some things needed to be changed...

Anyway, the mere fact that Nextcloud does not communicate about it anymore (2 years) despite a context which could be quite favorable to it, is for me the sign that the priority is not on this. At least for the moment.

I may try it once again to check but I have low expectations ;)

have a good day
